Classes/Domain/AbstractAggregateRoot.php

```diff
@@ -58,34 +58,24 @@ protected function __construct(string $identifier)
     /**
      * Apply an event to the current aggregate root
      *
-     * If the event aggregate identifier and name is not set the event
-     * is automatically set with the current aggregate identifier
-     * and name.
+     * If the event is an instance of AggregateEventInterface, the aggregate identifier will be set
      *
      * @param EventInterface $event
-     * @param array $metadata
-     * @return void
      * @api
      */
-    final public function recordThat(EventInterface $event, array $metadata = [])
+    final public function recordThat(EventInterface $event)
```

Comment:
True, but I'm a bit unsure about if it doesn't make more sense to still set the basic aggregate-related metadata (version, aggregateId, ...) inside the recordThat method directly.

ie. `uncommittedEvents` is an array of `EventWithMetadata`. Let's at least quickly recap the pros/cons...

Pro:
 - the aggregate is fully event transport/metadata agnostic
 - aggregate becomes easier to reimplement

Con:
 - The logic for versioning (the events) belongs to the aggregate
 - occuredAt timestamp gets shifted to the time the event is started processing for persisting
 - other aggregate relevant metadata can no longer be specified by an aggregate implementation itself

Please add any points you find